Google.PowerShell.IntegrationTests/Storage/Storage.GcsObject.Tests.ps1

```diff
@@ -207,6 +212,66 @@ Describe "New-GcsObject" {
         $obj.Metadata["Content-Type"] | Should Be "image/jpeg"
         Remove-GcsObject $obj
     }
+
+    It "should upload an empty folder" {
+        $TestFolder = Join-Path $TestDrive "TestFolder"
+
+        try {
+            if (-not (Test-Path $TestFolder))
+            {
+                New-Item -ItemType Directory -Path $TestFolder | Out-Null
+            }
+
+            $folder = New-GcsObject -Bucket $bucket -Folder $TestFolder -Force
+            $folderName = "TestFolder/"
+
+            $folder.Name | should be $folderName
+            $folder.Size | should be 0
+
+            $folderOnline = Get-GcsObject -Bucket $bucket -ObjectName $folderName
+            $folderOnline.Name | should be $folderName
+            $folderOnline.Size | should be 0
+        }
+        finally
+        {
+            Remove-Item $TestFolder -Recurse -Force -ErrorAction Ignore
+        }
+    }
+
+    It "should upload a folder with files and subfolders" {
+        $TestFolder = Join-Path $TestDrive "TestFolder"
+
+        try {
+            if (-not (Test-Path $TestFolder))
+            {
+                New-Item -ItemType Directory -Path $TestFolder | Out-Null
+            }
+
+            "Hello, world" | Out-File (Join-Path $TestFolder "world.txt")
+            "Hello, mars" | Out-File (Join-Path $TestFolder "mars.txt")
+            "Hello, jupiter" | Out-File (Join-Path $TestFolder "jupiter.txt")
+
+            $TestSubfolder = Join-Path $TestFolder "TestSubfolder"
+            New-Item -ItemType Directory -Path $TestSubfolder | Out-Null
+
+            "Hello, saturn" | Out-File (Join-Path $TestSubfolder "saturn.txt")
+            "Hello, pluto" | Out-File (Join-Path $TestSubfolder "pluto.txt")
+
+            $result = New-GcsObject -Bucket $bucket -Folder $TestFolder -Force
+
+            $result.Count | should be 7
+            $result.Name -contains "TestFolder/jupiter.txt" | should be $true
+            $result.Name -contains "TestFolder/mars.txt" | should be $true
+            $result.Name -contains "TestFolder/TestSubfolder/pluto.txt" | should be $true
+
+            $saturn = Get-GcsObject -Bucket $bucket -ObjectName "TestFolder/TestSubfolder/saturn.txt"
+            $saturn.ContentType | should be "text/plain"
```

Comment:
Could you also add a couple of test scenarios that use the `-Prefix` and `-Delimiter` parameters to `Get-GcsObject`?

`Get-GcsObject -Delimiter "/"` should not include pluto and saturn.
`Get-GcsObject -Prefix "Test" -Recurse` should include pluto and saturn.